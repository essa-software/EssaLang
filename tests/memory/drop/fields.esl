struct DropTest {
    id: u32;
    
    func __drop__(this) {
        print("Dropped: {}\n", this.id);   
    }
}

struct Foo {
    dt1: DropTest;
    dt2: DropTest;

    func new(): Foo {
        return Foo {
            dt1: DropTest { id: 1 },
            dt2: DropTest { id: 2 },
        };
    }
}

func main() {
    let foo = Foo::new();
    print("foo's members are still alive:\n");
    print("dt1={}\n", foo.dt1.id);
    print("dt2={}\n", foo.dt2.id);
}
