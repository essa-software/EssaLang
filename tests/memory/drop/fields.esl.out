foo's members are still alive:
dt1=1
dt2=2
Dropped: 2
Dropped: 1
